src/diffusers/pipelines/stochatic_karras_ve/pipeline_stochastic_karras.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+import torch
+
+from tqdm.auto import tqdm
+
+from ...models import UNet2DModel
+from ...pipeline_utils import DiffusionPipeline
+from ...schedulers import KarrasVeScheduler
+
+
+class KarrasVePipeline(DiffusionPipeline):
+    """
+    Stochastic sampling from Karras et al. [1] tailored to the Variance-Expanding (VE) models [2].
+    Use Algorithm 2 and the VE column of Table 1 from [1] for reference.
+
+    [1] Karras, Tero, et al. "Elucidating the Design Space of Diffusion-Based Generative Models." arXiv preprint arXiv:2206.00364 (2022).
+    [2] Song, Yang, et al. "Score-based generative modeling through stochastic differential equations." In Proc. ICLR (2021).
+    """
+
+    unet: UNet2DModel
+    scheduler: KarrasVeScheduler
+
+    def __init__(self, unet, scheduler):
+        super().__init__()
+        scheduler = scheduler.set_format("pt")
+        self.register_modules(unet=unet, scheduler=scheduler)
+
+    @torch.no_grad()
+    def __call__(self, batch_size=1, num_inference_steps=50, generator=None, torch_device=None, output_type="pil"):
+        if torch_device is None:
+            torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        img_size = self.unet.config.sample_size
+        shape = (batch_size, 3, img_size, img_size)
+
+        model = self.unet.to(torch_device)
+
+        # sample x_0 ~ N(0, sigma_0^2 * I)
+        sample = torch.randn(*shape) * self.scheduler.config.sigma_max
+        sample = sample.to(torch_device)
+
+        self.scheduler.set_timesteps(num_inference_steps)
+
+        for i in tqdm(self.scheduler.timesteps):
+            # here sigma_t == t_i from the paper
+            # sigmas are multiplied by 2 for compatibility with the original VE scheduler's sigma range
+            sigma = self.scheduler.schedule[i]
+            sigma_prev = self.scheduler.schedule[i - 1] if i > 0 else 0
+
+            # 1. Select temporarily increased noise level sigma_hat
+            # 2. Add new noise to move from sample_i to sample_hat
+            sample_hat, sigma_hat = self.scheduler.get_model_inputs(sample, sigma, generator=generator)
```

Comment:
`get_model_inputs` doesn't seem to describe the function well. Maybe let's call it `add_noise_to_input` or `churn_input`.

Reply:
I would just call it `get_sigmas(...)`, but I understand where `get_model_inputs` is coming from (to make the API as generic for all schedulers as possible which is also a good argument)

Here I would be in favor of being a bit more specific with something like `get_sigmas(...)` which is something most continuous schedulers have and which could become a common API across continuous schedulers. 
So I think here it makes sense to favor intuitive, readable code over easy-to-use. 

Also it scares me a bit to see that `sample` is an input to the function, it gives the impression that it's used to compute the sigmas. Can we instead maybe just pass `shape` and `device`?

Reply:
Note that `sample` is also used in computation: `sample_hat = sample + ((sigma_hat**2 - sigma**2)**0.5 * eps)`

Reply:
`get_sigmas` sound good to me, but still not clear enough, as it adds noise to the current sigmas rather than giving completly new `sigma`.

> Also it scares me a bit to see that sample is an input to the function, it gives the impression that it's used to compute the sigmas. Can we instead maybe just pass shape and device?

Good poin! In this case we will need to compute the `sample_hat` in the pipeline then. Also, the function then will have to return the `eps` as well which is needed to compute `sample_hat`.  I'm still in favor of `add_noise_to_input` as it makes clear that the function adds noise to the inputs rather using `sample` to compute `sigma`. That's also how the paper paper describes this part.

Reply:
Opting for `add_noise_to_input` for now :)

Reply:
ah sorry @anton-l you're right - missed that part. Then think we should more or less leave as is and maybe rename to `compute_sigmas(...)` to make sure reader knows that sigmas are dependent on the sample and completed (not taken from a predefined list)

Feel free to go ahead as you want though @anton-l